### PR TITLE
Cleanup child loggers

### PR DESCRIPTION
**Pull request by hexa00**

### CODE REVIEW

File: packages/process/src/node/process.ts [hunk 1]
```diff
@@ -57,26 +55,10 @@ export abstract class Process implements Disposable {
...
-            const p = new Promise<void>(resolve => {
-                this.kill();
-                this.exitEmitter.event(event => resolve());
-            });
-            p.then(() => {
-                this.exitEmitter.dispose();
-                this.errorEmitter.dispose();
-            });
-        } else {
-            this.exitEmitter.dispose();
```
Comment (akosyakov) [inline packages/process/src/node/process.ts]: Emitters don't seem to be disposed. Is not it required to remove references from listeners?

Comment (hexa00) [reply packages/process/src/node/process.ts]: As long as you clean the event handlers like:
in base-terminal-server.ts
```typescript
toDispose.push(term.onError(error => {
```
There's no more refs to the emitters and they do get cleaned-up see this diff between memory when a terminal is there and after I close it:
dipose-emitters

**Comment by akosyakov** [comment]
Is it still WIP?

File: packages/terminal/src/node/shell-process.ts [hunk 2]
```diff
@@ -41,7 +41,7 @@ export class ShellProcess extends TerminalProcess {
     constructor(
         @inject(ShellProcessOptions) options: ShellProcessOptions,
         @inject(ProcessManager) processManager: ProcessManager,
-        @inject(ILogger) logger: ILogger
+        @inject(ILogger) @named("package") logger: ILogger
```
Comment (akosyakov) [inline packages/terminal/src/node/shell-process.ts]: I think it should be a constant

just curious, why `package`?

**Comment by hexa00** [comment]
Ready for review

File: packages/terminal/src/node/shell-process.ts (hunk 2 above)
Comment (hexa00) [reply packages/terminal/src/node/shell-process.ts]: that was wrong... updated with the pkg name

**Pull request merged**